### PR TITLE
MINOR: bug fixes to ducktape services

### PR DESCRIPTION
**Pull request by alexlod**
Here's a (mostly successful) run with these changes:

http://testing.confluent.io/confluent-kafka-branch-builder-system-test-results/?prefix=2016-06-27--001.1467080884--alexlod--ducktape-fixes--ad85493/

At least one of the failed tests is failing in trunk, too:

http://testing.confluent.io/confluent-kafka-branch-builder-system-test-results/?prefix=2016-06-28--001.1467090978--alexlod--ducktape-fixes--ad85493/

The contribution is my original work and I license the work to the project under the project's open source license.

### CODE REVIEW

File: tests/kafkatest/services/security/minikdc.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def start_node(self, node):
         self.logger.info(props_file)
 
         kafka_principals = ' '.join(['kafka/' + kafka_node.account.hostname for kafka_node in self.kafka_nodes])
-        principals = 'client ' + kafka_principals + self.extra_principals
+        principals = 'client ' + kafka_principals + ' ' + self.extra_principals
```
Comment (ijuma) [inline tests/kafkatest/services/security/minikdc.py:68]: Why do we need this?

Comment (alexlod) [inline tests/kafkatest/services/security/minikdc.py:68]: `extra_principals` is currently never used. I found it confusing that I was expected to set `extra_principals` with a leading `' '`. This isn't needed, but IMO it cleans up the API.

File: tests/kafkatest/services/kafka/kafka.py [hunk 2]
```diff
@@ -38,8 +38,7 @@
 class KafkaService(KafkaPathResolverMixin, JmxMixin, Service):
 
     PERSISTENT_ROOT = "/mnt"
-    STDOUT_CAPTURE = os.path.join(PERSISTENT_ROOT, "kafka.log")
-    STDERR_CAPTURE = os.path.join(PERSISTENT_ROOT, "kafka.log")
+    STDOUT_STDERR_CAPTURE = os.path.join(PERSISTENT_ROOT, "server-start-stdout-stderr.log")
```
Comment (granders) [inline tests/kafkatest/services/kafka/kafka.py:41]: 👍

File: tests/kafkatest/services/security/minikdc.py (hunk 1 above)
Comment (granders) [inline tests/kafkatest/services/security/minikdc.py:68]: Agree, this seems cleaner

File: tests/kafkatest/services/zookeeper.py [hunk 3]
```diff
@@ -72,7 +72,8 @@ def start_node(self, node):
         self.logger.info(config_file)
         node.account.create_file("/mnt/zookeeper.properties", config_file)
 
-        start_cmd = "export KAFKA_OPTS=\"%s\";" % self.kafka_opts 
+        start_cmd = "export KAFKA_OPTS=\"%s\";" % (self.kafka_opts + ' ' + self.security_system_properties) \
```
Comment (granders) [inline tests/kafkatest/services/zookeeper.py:75]: where is `self.security_system_properties` defined/created?

**Comment by granders** [comment]
one minor question, but looks good to me

File: tests/kafkatest/services/kafka/kafka.py (hunk 2 above)
Comment (ijuma) [inline tests/kafkatest/services/kafka/kafka.py:41]: What is the reasoning for the file name? Does this only capture stdout and stderr for the server start?

Comment (alexlod) [inline tests/kafkatest/services/kafka/kafka.py:41]: @ijuma: it captures whatever shows up in stdout and stderr while Kafka is running. Without this change, any SASL or SSL debugging messages are lost. I'm happy to rename the file if you think it's misleading.

File: tests/kafkatest/services/zookeeper.py (hunk 3 above)
Comment (alexlod) [inline tests/kafkatest/services/zookeeper.py:75]: Line 51/52:

```
@property
def security_system_properties(self):
```

`security_system_properties` was defined but never used.

**Comment by alexlod** [comment]
I just added another commit with an additional minor change: changing `zk_set_acl` in `kafka.py` to a boolean, not a string boolean.

**Comment by granders** [comment]
LGTM
ping @ijuma or @ewencp for final review/commit

**Comment by ewencp** [comment]
LGTM, we have a test run that is currently only exhibiting other known transient failures.

**Comment by alexlod** [comment]
Any ideas why this was closed? I assume I should make a new PR?

**Comment by ijuma** [comment]
@alexlod, it was merged.

**Comment by alexlod** [comment]
Oh... great! Thanks @ijuma, @granders, @ewencp for reviewing.

**Pull request closed without merging**